source/simulator/stokes_matrix_free.cc

```diff
@@ -2264,13 +2274,22 @@ namespace aspect
                   for (unsigned int q=0; q<n_q_points; ++q)
                     {
                       for (unsigned int k=0; k<dofs_per_cell; ++k)
-                        symgrad_phi_u[k] = fe_values[velocities].symmetric_gradient (k, q);
+                        {
+                          symgrad_phi_u[k] = fe_values[velocities].symmetric_gradient (k, q);
+                          div_phi_u[k] = fe_values[velocities].divergence (k, q);
+                        }
 
                       const double JxW = fe_values.JxW(q);
                       for (unsigned int i=0; i<dofs_per_cell; ++i)
                         for (unsigned int j=0; j<dofs_per_cell; ++j)
-                          cell_matrix(i,j) += 2. * viscosity * (symgrad_phi_u[i]*symgrad_phi_u[j])
-                                              * JxW;
+                          {
+                            cell_matrix(i,j) += 2. * viscosity * (symgrad_phi_u[i]*symgrad_phi_u[j])
+                                                * JxW;
+
+                            if (is_compressible)
```

Comment:
Can you please put a comment block at the beginning of this function that contains the formula of what you are assembling?